### PR TITLE
FEATURE: QA-19307 Migrate test MultipleAttributeFilterManipulationTest.java from graphene

### PR DESCRIPTION
**Pull request by nknbich**
---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |
| `extended test - cypress - integrated <testName>`       | Run integrated tests against live backend                  |
| **Compatibility matrix test commands - TIGER Backend**  |                                                            |
| `extended test - matrix-test <AIO_version>`             | Run integrated tests against AIO versions.                 |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

`<AIO_version>` in commands is used to start test with multiple AIO instances - each instance in triggered by one jenkins build

-   To run with `master` and `stable`, execute command `extended test - matrix-test master,stable` or `extended test - matrix-test latest`
-   To run with specific version,ex: `2.3.0` and `2.3.1`, execute command `extended test - matrix-test 2.3.0,2.3.1`
-   In case `<AIO_version>` is empty, read versions from file `compTigerVersions.txt` of this repo

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `extended test - tiger-cypress - integrated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/14262/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/15819/ : FAILURE in 8m 05s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/14266/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/15821/ : FAILURE in 7m 27s

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

File: libs/sdk-ui-tests-e2e/cypress/tools/chart.ts [hunk 1]
```diff
@@ -131,4 +134,28 @@ export class Chart {
...
+    }
+
+    public assertShortenMetricNameInHeaderTable() {
+        this.assertShortenMetricName(TABLE_HEADER_TITLE, 590);
+    }
+
+    public assertShortenMetricName(element: string, maxWidth: number) {
+        cy.get(element).invoke("css", "text-overflow").should("equal", "ellipsis");
+        cy.get(element).invoke("width")
+            .should("be.lessThan", maxWidth);
```
Comment (NghiLamGD) [inline libs/sdk-ui-tests-e2e/cypress/tools/chart.ts:158]: use equal instead of lessThan, because maxWidth is a number you come up with yourself, it won't make any sense

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/14320/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/15864/ : SUCCESS in 12m 39s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/14322/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/15869/ : SUCCESS in 12m 13s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/14325/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/15872/ : SUCCESS in 12m 07s

File: libs/sdk-ui-tests-e2e/cypress/integration/01-sdk-ui-dashboard/dashboardShortenMetricName.spec.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+describe("Dashboard Shorten Metric Name", { tags: ["checklist_integrated_tiger"] }, () => {
+    beforeEach(() => {
+        Navigation.visit("dashboard/shorten-metric-name");
+    });
+
+    it("Table should shorten metric name", () => {
+        const table = new Widget(0).getTable();
+        table.waitLoaded();
+        table.assertShortenMetricName(496);
+        /*    });
```
Comment (NghiLamGD) [inline libs/sdk-ui-tests-e2e/cypress/integration/01-sdk-ui-dashboard/dashboardShortenMetricName.spec.ts:15]: why does these lines comment out?

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-tests-e2e/cypress/tools/chart.ts [hunk 3]
```diff
@@ -131,4 +133,24 @@ export class Chart {
             this.getElement().find(DATA_LABELS_SELECTOR).eq(i).contains(labels[i]).should("exist");
         }
     }
+
+    public assertShortenMetricNameInLegend(width: number) {
+        this.assertShortenMetricName(LEGEND_NAME_SELECTOR, width);
+    }
+
+    public hoverOnChart1stPointInHigh() {
```
Comment (NghiLamGD) [inline libs/sdk-ui-tests-e2e/cypress/tools/chart.ts:141]: can we use `clickSeriesPoint` instead?

File: libs/sdk-ui-tests-e2e/cypress/tools/chart.ts [hunk 4]
```diff
@@ -131,4 +133,24 @@ export class Chart {
             this.getElement().find(DATA_LABELS_SELECTOR).eq(i).contains(labels[i]).should("exist");
         }
     }
+
+    public assertShortenMetricNameInLegend(width: number) {
```
Comment (NghiLamGD) [inline libs/sdk-ui-tests-e2e/cypress/tools/chart.ts:137]: verify metric name in legend is an edge case, this function is no need, only assertShortenMetricName is enough, when you want to verify shorten metric in legend, just pass the css directly to the spec file

File: libs/sdk-ui-tests-e2e/cypress/tools/chart.ts [hunk 5]
```diff
@@ -131,4 +133,24 @@ export class Chart {
...
+        this.assertShortenMetricName(LEGEND_NAME_SELECTOR, width);
+    }
+
+    public hoverOnChart1stPointInHigh() {
+        cy.get(HIGHCHART_POINT_SELECTOR).first().as("firstElement");
+        cy.get("@firstElement").trigger("mouseover", { force: true });
+        return this;
+    }
+
+    public assertShortenMetricNameInTooltip(width: number) {
```
Comment (NghiLamGD) [inline libs/sdk-ui-tests-e2e/cypress/tools/chart.ts:147]: verify metric name in legend is an edge case, this function is no need, only assertShortenMetricName is enough, when you want to verify shorten metric in legend, just pass the css directly to the spec file

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/14335/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/15890/ : SUCCESS in 12m 17s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-tests-e2e/cypress/tools/chart.ts (hunk 4 above)
Comment (nknbich) [reply libs/sdk-ui-tests-e2e/cypress/tools/chart.ts]: I updated

File: libs/sdk-ui-tests-e2e/cypress/tools/chart.ts (hunk 5 above)
Comment (nknbich) [reply libs/sdk-ui-tests-e2e/cypress/tools/chart.ts]: I updated

File: libs/sdk-ui-tests-e2e/cypress/tools/chart.ts (hunk 3 above)
Comment (nknbich) [reply libs/sdk-ui-tests-e2e/cypress/tools/chart.ts]: Yes, it can, I updated

File: libs/sdk-ui-tests-e2e/cypress/integration/01-sdk-ui-dashboard/dashboardShortenMetricName.spec.ts (hunk 2 above)
Comment (nknbich) [reply libs/sdk-ui-tests-e2e/cypress/integration/01-sdk-ui-dashboard/dashboardShortenMetricName.spec.ts]: Typo, I updated

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/14338/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/15893/ : FAILURE in 9m 24s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-tests-e2e/cypress/integration/03-sdk-ui-charts/shortenMetricName.spec.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+import * as Navigation from "../../tools/navigation";
+import { Chart } from "../../tools/chart";
+import { Table } from "../../tools/table";
+
+describe("Shorten Metric Name", { tags: ["checklist_integrated_tiger"] }, () => {
+    it(`check shorten in legend and tooltip in chart`, () => {
+        Navigation.visit("visualizations/shortenmetricname/shorten-metric-name-chart-scenario");
+        const chart = new Chart(".s-column-chart");
+        chart.waitLoaded();
+        chart.assertShortenMetricName('.series-name', 176);
```
[the remark this answers is not in the document]
Comment (nknbich) [reply libs/sdk-ui-tests-e2e/cypress/integration/03-sdk-ui-charts/shortenMetricName.spec.ts]: yes I updated

File: libs/sdk-ui-tests-e2e/cypress/integration/03-sdk-ui-charts/shortenMetricName.spec.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
+// (C) 2021 GoodData Corporation
+import * as Navigation from "../../tools/navigation";
+import { Chart } from "../../tools/chart";
+import { Table } from "../../tools/table";
```
Comment (NghiLamGD) [inline libs/sdk-ui-tests-e2e/cypress/integration/03-sdk-ui-charts/shortenMetricName.spec.ts:4]: 1 empty line

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/14340/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/15895/ : FAILURE in 9m 35s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/14342/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/15897/ : FAILURE in 8m 22s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/14345/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/15900/ : SUCCESS in 12m 45s

**Comment by NghiLamGD** [comment]
LGTM

File: libs/sdk-ui-tests-e2e/scenarios/src/components/Scenarios/Dashboard/DashboardShortenMetricNameScenario.tsx [hunk 8]
```diff
@@ -0,0 +1,16 @@
+// (C) 2021-2022 GoodData Corporation
```
Comment (caohhung) [inline libs/sdk-ui-tests-e2e/scenarios/src/components/Scenarios/Dashboard/DashboardShortenMetricNameScenario.tsx:1]: `// (C) 2023 GoodData Corporation`

File: libs/sdk-ui-tests-e2e/scenarios/src/components/Scenarios/Visualizations/ShortenMetricName/ShortenMetricNameChartScenario.tsx [hunk 9]
```diff
@@ -0,0 +1,15 @@
+// (C) 2007-2023 GoodData Corporation
```
Comment (caohhung) [inline libs/sdk-ui-tests-e2e/scenarios/src/components/Scenarios/Visualizations/ShortenMetricName/ShortenMetricNameChartScenario.tsx:1]: `// (C) 2023 GoodData Corporation`

File: libs/sdk-ui-tests-e2e/scenarios/src/components/Scenarios/Visualizations/ShortenMetricName/ShortenMetricNameTableScenario.tsx [hunk 10]
```diff
@@ -0,0 +1,16 @@
+// (C) 2007-2023 GoodData Corporation
```
Comment (caohhung) [inline libs/sdk-ui-tests-e2e/scenarios/src/components/Scenarios/Visualizations/ShortenMetricName/ShortenMetricNameTableScenario.tsx:1]: `// (C) 2023 GoodData Corporation`

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-tests-e2e/scenarios/src/components/Scenarios/Dashboard/DashboardShortenMetricNameScenario.tsx (hunk 8 above)
Comment (nknbich) [reply libs/sdk-ui-tests-e2e/scenarios/src/components/Scenarios/Dashboard/DashboardShortenMetricNameScenario.tsx]: I updated

File: libs/sdk-ui-tests-e2e/scenarios/src/components/Scenarios/Visualizations/ShortenMetricName/ShortenMetricNameChartScenario.tsx (hunk 9 above)
Comment (nknbich) [reply libs/sdk-ui-tests-e2e/scenarios/src/components/Scenarios/Visualizations/ShortenMetricName/ShortenMetricNameChartScenario.tsx]: I updated

File: libs/sdk-ui-tests-e2e/scenarios/src/components/Scenarios/Visualizations/ShortenMetricName/ShortenMetricNameTableScenario.tsx (hunk 10 above)
Comment (nknbich) [reply libs/sdk-ui-tests-e2e/scenarios/src/components/Scenarios/Visualizations/ShortenMetricName/ShortenMetricNameTableScenario.tsx]: I updated

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/14357/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/15911/ : SUCCESS in 14m 51s

**Review by caohhung — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3843

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/2858/ : SUCCESS in 9m 50s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/15912/ : SUCCESS in 14m 53s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/418/ : SUCCESS in 14m 16s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/4014/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/739/ : FAILURE in 20s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/379/ : SUCCESS in 7m 57s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-compatibility-pipeline/243/ : FAILURE in 4m 41s

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3843

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/2944/ : SUCCESS in 9m 09s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/16182/ : SUCCESS in 12m 32s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/499/ : SUCCESS in 9m 58s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/4100/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/825/ : FAILURE in 1m 11s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/459/ : SUCCESS in 7m 59s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-compatibility-pipeline/328/ : FAILURE in 5m 33s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/14773/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/16332/ : FAILURE in 6m 56s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/14803/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/16369/ : SUCCESS in 7m 35s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/14804/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/16370/ : SUCCESS in 6m 57s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/14805/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/16371/ : SUCCESS in 7m 03s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/14806/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/16372/ : SUCCESS in 7m 00s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/14808/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/16374/ : SUCCESS in 6m 53s

**Comment by hungnguyenauto** [comment]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3843

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.
